### PR TITLE
Forge Bug #26 : Design Docker runtime in TOSCA. Initial commit

### PR DESCRIPTION
**Pull request by patduc38**
This component allows to install docker-ce on a compute node (optionally proxy configuration can be done as well)

### CODE REVIEW

File: org/ystia/docker/ansible/playbooks/create.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      repo: "{{debian_docker_repository}}"
+      state: present
+    when: ansible_os_family == 'Debian'
+
+  - name: Ensure Docker is installed
+    package:
+      name: "{{ docker_package_name }}"
+      state: present
+
+  - name: Enable munge service
```
Comment (gavillom) [inline org/ystia/docker/ansible/playbooks/create.yaml]: Enable Docker service, right ?

File: org/ystia/docker/ansible/playbooks/templates/http-proxy.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[Service]
+Environment="HTTP_PROXY={{ PROXY_URL }}"
+
```
Comment (gavillom) [inline org/ystia/docker/ansible/playbooks/templates/http-proxy.conf.j2]: HTTPS_PROXY should also be added

**Comment by gavillom** [comment]
Global remark : I would suggest to rely on proxy defined on the system rather than asking user to set its own proxy through a property. WDYT ? Because in our case and defined as it's right now, we may have troubles in installing a local Docker registry and trying to use it because Docker will  try to go through proxy to reach him.

**Comment by loicalbertin** [comment]
> Global remark : I would suggest to rely on proxy defined on the system rather than asking user to set its own proxy through a property. WDYT ? Because in our case and defined as it's right now, we may have troubles in installing a local Docker registry and trying to use it because Docker will try to go through proxy to reach him.

I think we had the exact same discussion here https://github.com/ystia/forge/pull/15

File: org/ystia/docker/ansible/playbooks/create.yaml (hunk 1 above)
Comment (patduc38) [reply org/ystia/docker/ansible/playbooks/create.yaml]: you're right, I will modify, thx

File: org/ystia/docker/ansible/playbooks/templates/http-proxy.conf.j2 (hunk 2 above)
Comment (patduc38) [reply org/ystia/docker/ansible/playbooks/templates/http-proxy.conf.j2]: Ok, I will rely on proxy env variables.

File: org/ystia/docker/ansible/types.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        default: ""
+        required: false
+        description: >
+          docker package version
+      repository_url:
+        type: string
+        default: ""
+        required: false
+        description: >
+          repository url to install docker from
```
[the remark this answers is not in the document]
Comment (patduc38) [reply org/ystia/docker/ansible/types.yml]: I will modify descriptions to indicate that these property are optional, and what are the default values

[the remark this answers is not in the document]
Comment (patduc38) [reply org/ystia/docker/ansible/types.yml]: I din't really get the point regarding the need of a public abstract component, could you precise why you think it should be better ?

**Comment by patduc38** [comment]
Thanks Mathis and Loic for your kind review and comments

**Pull request merged**